src/internet_identity/src/storage.rs

```diff
@@ -129,25 +206,27 @@ impl<T: candid::CandidType + serde::de::DeserializeOwned, M: Memory> Storage<T,
             ));
         }
 
-        if (id_range_hi - id_range_lo) > DEFAULT_RANGE_SIZE {
+        if (id_range_hi - id_range_lo) > DEFAULT_RANGE_SIZE_V1 {
             trap(&format!(
                 "id range [{}, {}) is too large for a single canister (max {} entries)",
-                id_range_lo, id_range_hi, DEFAULT_RANGE_SIZE,
+                id_range_lo, id_range_hi, DEFAULT_RANGE_SIZE_V1,
             ));
         }
 
         Self {
             header: Header {
                 magic: *b"IIC",
-                version: 1,
+                version: 3,
                 num_users: 0,
                 id_range_lo,
                 id_range_hi,
-                entry_size: DEFAULT_ENTRY_SIZE,
+                entry_size: DEFAULT_ENTRY_SIZE_V3,
                 salt: EMPTY_SALT,
+                entry_size_new: DEFAULT_ENTRY_SIZE_V3,
+                new_layout_start: 0,
```

Comment:
I think I misunderstood this. Shouldn't this start at the end of stable memory?

Reply:
This is the state after the migration being initialized directly.